crates/ruff/src/cache.rs

```diff
@@ -180,12 +180,22 @@ impl Cache {
             .write_all(&serialized)
             .context("Failed to write serialized cache to temporary file.")?;
 
-        temp_file.persist(&self.path).with_context(|| {
-            format!(
-                "Failed to rename temporary cache file to {}",
-                self.path.display()
-            )
-        })?;
+        if let Err(err) = temp_file.persist(&self.path) {
```

Comment:
The `open` method on this struct already uses a similar pattern of warning (but ignoring) some errors.